packages/contracts-bedrock/src/L1/SuperchainConfig.sol

```diff
@@ -92,4 +117,49 @@ contract SuperchainConfig is Initializable, ISemver {
         Storage.setAddress(GUARDIAN_SLOT, _guardian);
         emit ConfigUpdate(UpdateType.GUARDIAN, abi.encode(_guardian));
     }
+
+    /// @notice Adds a new chain to the dependency set.
+    ///         Adds the new chain as a dependency for all existing chains in the dependency set, and vice versa. It
+    ///         also stores the SystemConfig address of it, and authorizes the OptimismPortal on the SharedLockbox.
+    /// @param _chainId     The chain ID.
+    /// @param _systemConfig The address of the SystemConfig contract.
+    function addChain(uint256 _chainId, address _systemConfig) external {
+        // TODO: Updater role TBD, using guardian for now.
+        if (msg.sender != guardian()) revert Unauthorized();
+        // Add to the dependency set and check it is not already added (`add()` returns false if it already exists)
+        if (!_dependencySet.add(_chainId)) revert ChainAlreadyAdded();
+
+        systemConfigs[_chainId] = _systemConfig;
+
+        // If the dependency set is empty, there is no need to update the dependencies
+        if (_dependencySet.length() != 0) {
```

Comment:
Great catch. WIll remove it.

This issue arised bc I was using `length - 2` and we needed to check that length was `> 1`.
But since `length - 2` was off, we used `length - 1` and updated the check on the if to `!= 0` -- when I had to remove it.

Reply:
Addressed here https://github.com/defi-wonderland/optimism/pull/138/commits/15c9f13642696cb9772ca5bc168af5e57f5d09ab